tests/summ_ai_api/summ_ai_test.py

```diff
@@ -224,3 +477,147 @@ async def test_summ_ai_error_handling(
+    """
+    response_data_translation_succeeded = {
+        "dummy": "dummy",
+        "translated_text": "Das ist einfaches Deutsch.",
+    }
+    assert (
+        SummAiApiClient(MockedRequest(), "form_dummy").validate_response(
+            response_data_translation_succeeded, 200
+        )
+        is True
+    ), "if translated_text found in reponse-data, validate_response should return True"
+
+    response_data_translation_failed = {
+        "dummy": "dummy",
+        "error": "no translation test error",
+    }
+    with pytest.raises(SummAiRuntimeError):
+        SummAiApiClient(MockedRequest(), "form_dummy").validate_response(
+            response_data_translation_failed, 200
+        )
+
+
+def test_check_rate_limit_exceeded(settings, caplog):
+    """
+    Test for check_rate_limit_exceeded method. Tests if return is True when
+    response-status = 429 (too many requests) or 529 (site overloaded),
+    otherwise False
+
+    :param settings: The Django settings
+    :type settings: :fixture:`settings`
+
+    :param caplog: Fixture for asserting log messages in tests (see :fixture:`pytest:caplog`)
+    :type caplog: ~_pytest.logging.LogCaptureFixture
+    """
+
+    assert (
+        SummAiApiClient(MockedRequest(), "form_dummy").check_rate_limit_exceeded(200)
+        is False
+    ), "if response-status not in (429, 529), check-rate-limit should return False"
+    with pytest.raises(SummAiRateLimitingExceeded):
+        SummAiApiClient(MockedRequest(), "form_dummy").check_rate_limit_exceeded(429)
+    with pytest.raises(SummAiRateLimitingExceeded):
+        SummAiApiClient(MockedRequest(), "form_dummy").check_rate_limit_exceeded(529)
+
+    errors = tuple(
+        map(
+            lambda record: record.message,
+            filter(lambda record: record.levelname == "ERROR", caplog.records),
+        )
+    )
+    assert len(errors) >= 2
+    assert (
+        f"SUMM.AI translation is waiting for {settings.SUMM_AI_RATE_LIMIT_COOLDOWN}s because the rate limit has been exceeded"
+        in errors
+    ), "after check_rate_limit_exceeded() with 429 or 529, a logging entry should appear"
+
+    assert (
+        SummAiApiClient(MockedRequest(), "form_dummy").check_rate_limit_exceeded(500)
+        is False
+    ), "if response-status not in (429, 529), check-rate-limit should return False"
+
+
+def test_patient_task_queue_normal_deque():
+    """
+    Test for PatientTaskQueue Class. Tests that a first-in/first out (fifo) deque-queue is build from a task-list
+    Checks that next() removes one element each from the queue
+    """
+
+    tasks = [1, 2, 3]
+    task_generator = PatientTaskQueue(tasks, 1)
+
+    # we check that list comes reverse (fifo-queue)
+    for i in range(len(task_generator)):
+        assert next(task_generator) == tasks[i]
+
+    assert len(task_generator) == 0
+    # we check that next() did remove elements from the queue
+    with pytest.raises(StopIteration):
+        next(task_generator)
```

Comment:
```suggestion
async def test_patient_task_queue_normal_deque():
    """
    Test for PatientTaskQueue Class. Tests that a first-in/first out (fifo) deque-queue is build from a task-list
    Checks that __anext__() removes one element each from the queue
    """

    tasks = [1, 2, 3]
    task_generator = PatientTaskQueue(tasks, 1)

    # we check that list comes reverse (fifo-queue)
    i = 0
    async for task in task_generator:
        assert task == tasks[i]
        i += 1

    assert len(task_generator) == 0
    # we check that __anext__() did remove elements from the queue
    with pytest.raises(StopAsyncIteration):
        await task_generator.__anext__()
```
If we're using `PatientTaskQueue` as asynchronous iterator, we should also test the async iter features instead of the sync ones.